### PR TITLE
emitter ut framework and one unit test

### PR DESCRIPTION
**Pull request by chunyu3**
# Description

- emitter UT framework and one unit test

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: src/CADL.Extension/Emitter.Csharp/package.json [hunk 1]
```diff
@@ -24,24 +24,26 @@
         "watch": "tsc -p . --watch",
         "pack": "npx webpack",
         "lint-cadl-library": "cadl compile . --warn-as-error --import @cadl-lang/library-linter --no-emit",
-        "test": "mocha",
+        "test": "mocha dist/test/**/*.js",
         "test-official": "c8 mocha --forbid-only",
         "lint": "eslint . --ext .ts --max-warnings=0",
         "lint:fix": "eslint . --fix --ext .ts",
-        "prettier:fix": "prettier --write --config .prettierrc.json ./src/**/*.ts",
+        "prettier:fix": "prettier --write --config .prettierrc.json ./src/**/*.ts ./test/**/*.ts",
```
Comment (pshao25) [inline src/CADL.Extension/Emitter.Csharp/package.json:31]: Put this to "prettier" as well.

File: src/CADL.Extension/Emitter.Csharp/test/Unit/propertyType.test.ts [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+    EmitContext,
+    isGlobalNamespace,
+    Namespace,
+    navigateTypesInNamespace,
+    Program,
+    Type
+} from "@cadl-lang/compiler";
+import { getInputType } from "../../src/lib/model.js";
+
+describe("compiler : array", () => {
```
Comment (pshao25) [inline src/CADL.Extension/Emitter.Csharp/test/Unit/propertyType.test.ts:31]: As a unit test, let's make the description to contain:
1. which method is testing
2. what is the scenario/expectation

File: src/CADL.Extension/Emitter.Csharp/package.json (hunk 1 above)
Comment (chunyu3) [reply src/CADL.Extension/Emitter.Csharp/package.json:31]: Thanks @pshao25  Good suggestion. I will add.

File: src/CADL.Extension/Emitter.Csharp/test/Unit/propertyType.test.ts (hunk 2 above)
Comment (chunyu3) [reply src/CADL.Extension/Emitter.Csharp/test/Unit/propertyType.test.ts]: I updated the description.

**Review by pshao25 — APPROVED** [review]

**Pull request merged**